client/post-editor/post-editor.jsx

```diff
@@ -379,6 +379,13 @@ const PostEditor = React.createClass( {
 			this.setState( this.getInitialState(), function() {
 				this.refs.editor.setEditorContent( '' );
 			} );
+		} else if ( this.state.isNew && this.state.hasContent && ! this.state.isDirty ) {
+			// Is a copy of an existing post.
+			// When copying a post, the created draft is new and the editor is not yet dirty, but it already has content.
```

Comment:
Ok, this is quite hard to track down so I may actually be wrong, even though extensive testing seems to support my assumption.

(note: line numbers may differ, I'm looking around in this branch)

Flux state `isDirty` is this [(lib/posts/post-edit-store.js#L453):](https://github.com/Automattic/wp-calypso/blob/add/copy-post-rebase-example/client/lib/posts/post-edit-store.js#L453)
```js
_post !== _savedPost || _rawContent !== _initialRawContent
```
The first check is inevitably correct: during its init, a copy is not yet saved.
So let's see the second one.

`_rawContent` is only set by `setRawContent` ([lib/posts/post-edit-store.js#L216](https://github.com/Automattic/wp-calypso/blob/add/copy-post-rebase-example/client/lib/posts/post-edit-store.js#L216)) which, in this case, is called by `saveRawContent` ([client/post-editor/post-editor.jsx#L429](https://github.com/Automattic/wp-calypso/blob/add/copy-post-rebase-example/client/post-editor/post-editor.jsx#L429)).

`saveRawContent` is triggered (and debounced by 200ms) by editor changes, which in turn are triggered by `this.refs.editor.setEditorContent` ([client/post-editor/post-editor.jsx#L387](https://github.com/Automattic/wp-calypso/blob/add/copy-post-rebase-example/client/post-editor/post-editor.jsx#L387)).

So what happens: on load, `actions.startEditingNew` and `actions.edit` [(lib/posts/actions.js#L162)](https://github.com/Automattic/wp-calypso/blob/add/copy-post-rebase-example/client/lib/posts/actions.js#L162) set the post attributes in the store.
The editor - which is not dirty yet! - receives the changes in the store, checks that `isNew && hasContent && ! isDirty` and proceeds setting its own content. Once debounced, `setRawContent` sets both the content and `isDirty = true`, preventing additional updates to the editor content.